src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -149,13 +151,19 @@ await ProcessPendingUpdatesAsync(
     /// <returns>
     ///     True if updates have been applied; <see langword="false" /> otherwise.
     /// </returns>
-    public async Task ProcessPendingUpdatesAsync(SubscriptionUpdateWorkItem update, bool forceApply)
+    public async Task ProcessPendingUpdatesAsync(SubscriptionUpdateWorkItem update, bool forceApply, BuildDTO? build)
     {
         _logger.LogInformation("Processing pending updates for subscription {subscriptionId}", update.SubscriptionId);
         bool isCodeFlow = update.SubscriptionType == SubscriptionType.DependenciesAndSources;
         InProgressPullRequest? pr = await _pullRequestState.TryGetStateAsync();
         PullRequest? prInfo;
 
+        if (build == null)
+        {
+            _logger.LogInformation("Looking up build {buildId}", update.BuildId);
+            build = await _sqlClient.GetBuildAsync(update.BuildId);
+        }
```

Comment:
Same here, I'd remove the logging:

```suggestion
        build ??= await _sqlClient.GetBuildAsync(update.BuildId);
```